### PR TITLE
[Prim][PIR] support clip op forward in prim pir

### PR DESCRIPTION
**Pull request by Eddie-Wang1120**
### PR Category

Others

### PR Types

Others

### Description
[Prim][PIR] support clip op forward in prim pir
Paddle/test/ir/pir/cinn/symbolic/test_sub_graph_stable_diffusion_3_st.py  Paddle/test/ir/pir/cinn/sub_graphs/test_sub_graph_79.py
这两个单侧with_cinn=False后都可以通过
PR-CI-CINN报错详细信息
test_sub_graph_stable_diffusion_3_st
image
test_sub_graph_79
image

### CODE REVIEW

File: paddle/fluid/primitive/composite/composite.h [hunk 1]
```diff
@@ -986,6 +986,25 @@ std::tuple<Tensor, Tensor> flatten_decomp(const Tensor& x,
...
+  if (has_dynamic_shape(x.shape())) {
+    min_reshape = backend::expand_with_tensor<T>(
+        cast<T>(min, DataType::FLOAT64), shape<T>(x));
+    max_reshape = backend::expand_with_tensor<T>(
+        cast<T>(max, DataType::FLOAT64), shape<T>(x));
+  } else {
+    min_reshape = expand<T>(cast<T>(min, DataType::FLOAT64), x.shape());
+    max_reshape = expand<T>(cast<T>(max, DataType::FLOAT64), x.shape());
+  }
+  auto ans = maximum<T>(minimum<T>(cast<T>(x, DataType::FLOAT64), max_reshape),
```
Comment (cyber-pioneer) [inline paddle/fluid/primitive/composite/composite.h]: dtype的来源确认一下哈

Comment (Eddie-Wang1120) [reply paddle/fluid/primitive/composite/composite.h]: 好嘞好嘞我晚上弄一下

Comment (Eddie-Wang1120) [reply paddle/fluid/primitive/composite/composite.h]: done

**Comment by Eddie-Wang1120** [comment]
Paddle/test/ir/pir/cinn/symbolic/test_sub_graph_stable_diffusion_3_st.py  Paddle/test/ir/pir/cinn/sub_graphs/test_sub_graph_79.py
这两个单侧with_cinn=False后都可以通过
PR-CI-CINN报错详细信息
test_sub_graph_stable_diffusion_3_st
image
test_sub_graph_79
image

File: paddle/fluid/primitive/composite/composite.h [hunk 2]
```diff
@@ -986,6 +986,31 @@ std::tuple<Tensor, Tensor> flatten_decomp(const Tensor& x,
...
+  } else {
+    min_reshape = expand<T>(min, x.shape());
+    max_reshape = expand<T>(max, x.shape());
+  }
+
+  auto min_expand = cast<T>(min_reshape, x_promote.dtype());
+  auto max_expand = cast<T>(max_reshape, x_promote.dtype());
+
+  auto ans = maximum<T>(minimum<T>(x_promote, max_expand), min_expand);
+  return cast<T>(ans, x.dtype());
```
Comment (cyber-pioneer) [inline paddle/fluid/primitive/composite/composite.h:1011]: cast逻辑需要判断一下，半精度场景才需要

File: paddle/fluid/primitive/composite/composite.h [hunk 3]
```diff
@@ -986,6 +986,31 @@ std::tuple<Tensor, Tensor> flatten_decomp(const Tensor& x,
   }
 }
 
+template <typename T>
+Tensor clip_decomp(const Tensor& x, const Tensor& min, const Tensor& max) {
+  auto x_promote = x;
+  if (is_half_dtype(x.dtype())) {
+    x_promote = cast<T>(x, DataType::FLOAT32);
```
Comment (cyber-pioneer) [inline paddle/fluid/primitive/composite/composite.h]: clip不涉及密集计算，不需要类型提升

Comment (Eddie-Wang1120) [reply paddle/fluid/primitive/composite/composite.h]: done

**Pull request closed without merging**